daemon/src/battery/tplink.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{battery::BatteryState, error::RayhunterError};
+
+pub async fn get_battery_state() -> Result<BatteryState, RayhunterError> {
+    let uci_battery = tokio::process::Command::new("uci")
+        .arg("get")
+        .arg("battery.battery_mgr.power_level")
+        .output()
+        .await?;
+
+    let uci_plugged_in = tokio::process::Command::new("uci")
+        .arg("get")
+        .arg("battery.battery_mgr.is_charging")
+        .output()
+        .await?;
+
+    if !uci_battery.status.success() {
+        return Err(RayhunterError::BatteryLevelParseError);
+    }
+
+    if !uci_plugged_in.status.success() {
+        return Err(RayhunterError::BatteryPluggedInStatusParseError);
+    }
+
+    let uci_battery = String::from_utf8_lossy(&uci_battery.stdout)
+        .to_string()
+        .trim_end()
+        .parse()
+        .or(Err(RayhunterError::BatteryLevelParseError));
```

Comment:
this is more common:


```suggestion
        .map_err(RayhunterError::BatteryLevelParseError);
```

and I think you can use `?` here instead of further down.

Reply:
This isn't possible because `BatteryLevelParseError` is an enumeration, not a constructible error.

Reply:
I'll write it as a closure, just getting used to rust and took the easy way